lib/puppet/application.rb

```diff
@@ -420,12 +420,16 @@ def set_log_level(opts = nil)
   def handle_logdest_arg(arg)
     return if arg.nil?
 
-    begin
-      Puppet[:logdest] = arg
-      Puppet::Util::Log.newdestination(arg)
-      options[:setdest] = true
-    rescue => detail
-      Puppet.log_and_raise(detail, _("Could not set logdest to %{dest}.") % { dest: arg })
+    logdest = arg.split(',').map!(&:strip)
+    Puppet[:logdest] = arg
+
+    logdest.each do |dest|
+      begin
+        Puppet::Util::Log.newdestination(dest)
+        options[:setdest] = true
```

Comment:
This is because I worked on embedded lately, may not be useful here:
```suggestion
        options[:setdest] = true unless options[:setdest]
```